lvsfunc/util.py

```diff
@@ -102,6 +103,45 @@ def get_prop(frame: vs.VideoFrame, key: str, t: Type[T]) -> T:
     return prop
 
 
+def select_frames(clips: Union[vs.VideoNode, List[vs.VideoNode]],
+                  indicies: Union[List[int], List[Tuple[int, int]]]) -> vs.VideoNode:
+    """
+    Select frames from one or more clips at specified indices.
+
+    :param clips:   A clip or a list of clips to select the frames from
+    :param indexes: Indices of frames to select.
+
+                    Provide a list of indices for a single clip, or for multiple clips, a list of tuples in the
+                    form ``(clip_index, frame_index)``
+
+    :return:        The selected frames in a single clip
+    """
+    clips = [clips] if isinstance(clips, vs.VideoNode) else clips
+    for pos, index in enumerate(indicies):
+        if isinstance(index, int):
+            indicies[pos] = (0, index)
```

Comment:
in addition to the mypy issues, this is bad because mutating user input is bad

Reply:
Whilst mutating the userinput is probably not the best idea, I still don't like mypy telling me what I can and can't do >:(
Creating a copy of the list would probably a solution

Reply:
honestly i'd just throw this entire logic into the frameeval, it's not going to impact performance at all